subgraphs/hop-protocol/protocols/hop-protocol/src/L2/amm.ts

```diff
@@ -0,0 +1,356 @@
+    )[0];
+
+    const inputTokenTwo = NetworkConfigs.getTokenAddressFromPoolAddress(
+      event.address.toHexString()
+    )[1];
+    const poolConfig = NetworkConfigs.getPoolDetails(
+      event.address.toHexString()
+    );
+
+    const poolName = poolConfig[1];
+    const poolSymbol = poolConfig[0];
+    const hPoolName = poolConfig[2];
+
+    const sdk = SDK.initializeFromEvent(
+      conf,
+      new Pricer(),
+      new TokenInit(),
+      event
+    );
+
+    const pool = sdk.Pools.loadPool<string>(event.address);
+    const hPool = sdk.Pools.loadPool<string>(
+      Bytes.fromHexString(event.address.toHexString().concat("-").concat("1"))
+    );
+
+    const tokenOne = sdk.Tokens.getOrCreateToken(
+      Address.fromString(inputTokenOne)
+    );
+    const tokenTwo = sdk.Tokens.getOrCreateToken(
+      Address.fromString(inputTokenTwo)
+    );
+
+    sdk.Accounts.loadAccount(event.params.buyer);
+
+    if (!pool.isInitialized) {
+      pool.initialize(poolName, poolSymbol, BridgePoolType.LIQUIDITY, tokenOne);
+    }
+    if (!hPool.isInitialized) {
+      hPool.initialize(
+        hPoolName,
+        poolSymbol,
+        BridgePoolType.LIQUIDITY,
+        tokenTwo
+      );
+    }
+
+    updateAMMTVE(event.address, Address.fromBytes(tokenOne.id), hPool, pool);
+
+    pool.pool.relation = hPool.getBytesID();
+    hPool.pool.relation = hPool.getBytesID();
+
+    pool.addRevenueNative(BigInt.zero(), fees);
+  }
+}
+
+export function handleAddLiquidity(event: AddLiquidity): void {
+  if (!NetworkConfigs.getPoolsList().includes(event.address.toHexString())) {
+    log.error("Missing Config", []);
+    return;
+  }
+  const amount = event.params.tokenAmounts;
+  if (amount.length == 0) {
+    return;
+  }
+  const liquidity = amount[0].plus(amount[1]);
+
+  const inputTokenOne = NetworkConfigs.getTokenAddressFromPoolAddress(
+    event.address.toHexString()
+  )[0];
+
+  const inputTokenTwo = NetworkConfigs.getTokenAddressFromPoolAddress(
+    event.address.toHexString()
+  )[1];
+
+  const poolConfig = NetworkConfigs.getPoolDetails(event.address.toHexString());
+
+  const poolName = poolConfig[1];
+  const hPoolName = poolConfig[2];
+  const poolSymbol = poolConfig[0];
+
+  const sdk = SDK.initializeFromEvent(
+    conf,
+    new Pricer(),
+    new TokenInit(),
+    event
+  );
+
+  const pool = sdk.Pools.loadPool<string>(event.address);
+  const token = sdk.Tokens.getOrCreateToken(Address.fromString(inputTokenOne));
+  const hToken = sdk.Tokens.getOrCreateToken(Address.fromString(inputTokenTwo));
+  const acc = sdk.Accounts.loadAccount(event.params.provider);
+  const hPool = sdk.Pools.loadPool<string>(
+    Bytes.fromHexString(event.address.toHexString().concat("-").concat("1"))
+  );
+
+  if (!pool.isInitialized) {
+    pool.initialize(poolName, poolSymbol, BridgePoolType.LIQUIDITY, token);
+  }
+  if (!hPool.isInitialized) {
+    hPool.initialize(hPoolName, poolSymbol, BridgePoolType.LIQUIDITY, hToken);
+  }
+
+  if (
```

Comment:
Can you add a comment as to why this transaction needs special care?

Reply:
Okay